pennylane/_device.py

```diff
@@ -52,8 +52,9 @@ class Device(abc.ABC):
     _circuits = {}  #: dict[str->Circuit]: circuit templates associated with this API class
     _asarray = staticmethod(np.asarray)
 
-    def __init__(self, wires=1, shots=1000):
+    def __init__(self, wires=1, shots=1000, analytic=True):
```

Comment:
This was something that came up previously, and we made the decision to not include `analytic` in the base Device class (see https://github.com/XanaduAI/pennylane/pull/317/files#r325266577)

Reply:
(was there something here that necessitated adding this attribute?)

Reply:
Oh, that makes sense! This was mainly added such that the `device.analytic` attribute can be checked later (and that separate cases can be created for the default step size). From what I experienced, when the step size was increased to a higher value in the `analytic=True` case (such as `0.3`), tests for the gradient computation had bigger deviations and were failing. Would there be perhaps a better way around this?

Reply:
Oh, is this due to a testing issue? In tests, you are testing using a mocked out Device (where analytic doesn't exist) compared to a mocked out Default qubit?

Reply:
Yes, several tests fail (mostly from the CV case, but there are a couple of qubit based ones using the `qubit_device_2_wires` fixture from `conftest.py`, ~90 test cases affected).

Another alternative I gave a try was adding the possibility to set the step size only in the `analytic=False` case (this was using the condition of `and not self.device.analytic` on line 207). This way, as you were suggesting:
* only the mocked out `Device` objects would require an `analytic` attribute (which could be made up for in the tests, ~10 test cases are affected overall)
* we wouldn't have to add the `analytic` attribute in the `Device` class
* however, the step size would not be settable in the `analytic=True` case

From my discussion with @trbromley, it seemed that the ability to set the step size  in the `analytic=True` case would not be too crucial, but rather something nice to have.

(Sidenote: the condition of `and not self.device.analytic` on line 207 was mistakenly left in for now, since we have a branching logic for the default values based on `device.analytic`, this should be removed using this logic.)